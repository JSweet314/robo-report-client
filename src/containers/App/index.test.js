import React from "react";
import { shallow } from 'enzyme';
import { App, mapStateToProps } from './index';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App isLoggedIn={true} />);
  });

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    const mockState = { isLoggedIn: false };
    const mapped = mapStateToProps(mockState);

    it('should return isLoggedIn from the state', () => {
      expect(mapped).toHaveProperty('isLoggedIn', false);
    });
  });
});