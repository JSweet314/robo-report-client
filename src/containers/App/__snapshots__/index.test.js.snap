// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match a snapshot 1`] = `
<div
  className="app"
>
  <Header
    handleOAuthSignIn={[Function]}
    user={
      Object {
        "email": "walt@aol.com",
        "id": 1,
        "name": "Walter",
      }
    }
  />
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      component={[Function]}
      path="/welcomeNewUser"
    />
    <Route
      component={[Function]}
      path="/newComplaint"
    />
    <Route
      component={[Function]}
      path="/myReports"
    />
    <Route
      path="/myAccount"
      render={[Function]}
    />
  </Switch>
</div>
`;
